src/brod_group_subscriber.erl

```diff
@@ -35,7 +35,9 @@
 %%%=============================================================================
 
 -module(brod_group_subscriber).
+
 -behaviour(gen_server).
+-behaviour(brod_group_controller).
```

Comment:
Behaviour name must be brod_group_subscriber.

I suggest to rename this module to brod_group_subscriber_impl and move behaviour callbacks definition to a new brod_group_subscriber.erl.